apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2201,7 +2313,27 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = TriggersHandler.register_target(trigger_target)
 
     next_device_triggers = Map.put(device_triggers, event_type, new_targets)
+
     Map.put(state, :device_triggers, next_device_triggers)
+    |> maybe_cache_trigger_policy(trigger_target)
+  end
+
+  # TODO: consider what we should to with the cached policy if/when we allow updating a policy
+  defp maybe_cache_trigger_policy(%State{realm: realm_name} = state, %AMQPTriggerTarget{
+         parent_trigger_id: parent_trigger_id
+       }) do
+    case PolicyQueries.retrieve_policy_name(
+           realm_name,
+           parent_trigger_id
+         ) do
+      {:ok, policy_name} ->
+        Map.put(state, :trigger_id_to_policy_name, %{
+          parent_trigger_id => policy_name
+        })
```

Comment:
Since the state is a struct, use the map update syntax to have additional assurances against typos and/or missing keys.
Also, I think you're actually caching just the last trigger because you're overwriting the whole map.
```elixir
next_trigger_id_to_policy_name = Map.put(state.trigger_id_to_policy_name, parent_trigger_id, policy_name)
%{state | trigger_id_to_policy_name: next_trigger_id_to_policy_name}
```